### PR TITLE
MicrosoftSQL to Oracle - Changes synchronization

### LINKED ISSUE #147: MicrosoftSQL to Oracle - ongoing data replication
1. Have a table on MicrosoftSQL with timestamps.
2. Insert data into oracle via second table and MERGE query.

Jefim has an example for this.

### PR DESCRIPTION
**Pull request by jannejjj**
closes #147

Checklist for review:
- [x] Process name is fine, uses normal casing and descriptive
  - [x] If applicable, the process name follows the naming convention "{source/format} to {destination/format} - {object}"
- [x] Process description is fine, uses normal casing, dots and is descriptive
  - [x] Element names use normal casing, no unneeded capital letters
- [x] Correct casing used for formats and terms, e.g. CSV, FTP, JSON, Avro
  - [x] Element names do not contain dots
  - [x] Comments / note element text are in good language and end with dots
- [x] Template parameters checked
  - [x] Parameter follows the CamelCase naming convention, abbreviations included, e.g. ConnectionString, BaseUrl, ApiKey
  - [x] Parameter descriptions are good and end with dots
  - [x] Sensitive parameters are marked as secrets
- [x] "Turn off logging" switch is on for any task that has sensitive data as output, e.g. tokens in the header.
- [x] Process end points like error throws and return statements have titles (double-click) and reasonable information.
  - [x] Results have good summaries, preferably with statistics of some sort like "processed X files"
  - [x] Errors have some error info, usually from task (e.g. some ErrorMessage property contains additional error info)
  - [x] Errors are pointed downwards from their respective source, e.g. an exclusive decision
- [x] Process is organised with groups (dashed boxes)
- [x] Straight lines are preferred where possible
- [x] Subprocesses are forbidden
- [x] Text mode is preferred over Expressions in expressions, e.g. instead of *$"This is {#var.error}"* use *This is {{#var.error}}*
- [x] Process metadata file is present and content is fine (version number and name)
- [x] Process export file is present (process.json)
- [x] Process description file is present (all sections filled in)
- [x] Process description has a SVG screenshot of the process
- [x] Template files are going to the correct folder
- [x] Markdown paragraphs are separated with two newlines, so there is a blank line inbetween two paragraphs
- [x] If some system is integrated using e.g. a HTTP task (meaning we do not use a native connector for the system), then the system should be mentioned in additional systems used in metadata.json
- [x] If the template is handling some sample JSON data - make sure that the sample data is included in the long description (just one-two elements is enough)
- [x] If the template is assuming that there is a database table ready for data insertion / updates - include the table structure in the long description as a `CREATE TABLE` scipt.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new template for data synchronization between Microsoft SQL Server and Oracle databases.
	- Enables efficient fetching and processing of new or updated rows from Microsoft SQL to Oracle in manageable chunks.
	- Supports configurable parameters for enhanced flexibility, including connection strings, table names, and error handling.
	- Includes error logging and cleanup mechanisms to ensure smooth operations and data consistency.

- **Documentation**
	- Added metadata and process description files to provide foundational information about the synchronization template and its functionalities.

### CODE REVIEW

File: Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+This template process fetches new or updated rows from a table in a **MicrosoftSQL** database and inserts or updates the rows into a table in an **Oracle** database in chunks. This template process is meant to be used in conjunction with the "MicrosoftSQL to Oracle - Batch load" template, which can used initially to load data from the MicrosoftSQL table to the Oracle table.
```
Comment (charlotta-frends) [inline Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md:1]: Typo: which can [be] used initially to load data

File: Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/metadata.json
Comment (charlotta-frends) [inline Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/metadata.json:1]: Change the name to "Changes synchronization" to match the template name in the other files.

File: Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md
Comment (charlotta-frends) [inline Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md:1]: As with the other table, the example CREATE TABLE scripts should be added.

**Review by charlotta-frends — CHANGES_REQUESTED** [review]
Does cleaning up the staging table work as intended? Running the process does not show any errors, but the table seems to still have some rows in it when it should not (at least with my testing).

Additionally, now if the staging is successful but the merging results in an exception, the staging table is never cleaned, so should the template's error handling be altered somehow?

File: Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md
Comment (jannejjj) [reply Templates/Databases/MicrosoftSQL/MicrosoftSQL to Oracle - Changes sync/long-description.md:1]: Added example scripts. 👍

**Comment by jannejjj** [comment]
> Does cleaning up the staging table work as intended? Running the process does not show any errors, but the table seems to still have some rows in it when it should not (at least with my testing).

The TRUNCATE TABLE command was missing the schema name. 😄

> Additionally, now if the staging is successful but the merging results in an exception, the staging table is never cleaned, so should the template's error handling be altered somehow?

I changed it so that the merge and cleanup always happen, and the error check is after them.

**Review by charlotta-frends — APPROVED** [review]

**Pull request closed without merging**